common/races.h

```diff
@@ -860,6 +860,7 @@ uint16 GetRaceIDFromPlayerRaceBit(uint32 player_race_bit);
 
 float GetRaceGenderDefaultHeight(int race, int gender);
 
+bool IsGuard(uint16 race_id, uint8 texture_value, uint32 primaryfaction);
```

Comment:
`IsGuard` is a great function, but it should live in the `NPC` class and should not need any parameters passed to it because everything you need will be scoped within NPC

Your function becomes this (We also need constants in place of the magic numbers for factions)

```cpp
bool NPC::IsGuard()
{
	switch (GetRace()) {
		case RT_GUARD:
			if (GetTexture() == 1 || GetTexture() == 2)
				return true;
			break;
		case RT_IKSAR_2:
			if (GetTexture() == 1)
				return true;
			break;
		case RT_GUARD_2:
		case RT_GUARD_3:
		case RT_GUARD_4:
		case RT_HUMAN_3:
		case RT_HALFLING_2:
		case RT_ERUDITE_2:
		case RT_BARBARIAN_2:
		case RT_DARK_ELF_2:
		case RT_TROLL_2:
		case OGGOK_CITIZEN:
		case RT_DWARF_2:
			return true;
		default:
			break;
	}
	if (GetPrimaryFaction() == 255 || GetPrimaryFaction() == 265 || GetPrimaryFaction() == 333) { //these 3 factions of guards use player races instead of their own races so we must define them by faction.
		return true;
	}
	return false;
}
```

Your utilization of the function would simply end up being `IsGuard()` instead of passing parameters

Because the function becomes NPC scoped, you will either need to cast Mob to NPC (NPC inherits Mob, similarly to Client inheriting Mob) or you will need to explicitly loop through NPC's instead of Mobs and you will already have an NPC type without having to cast

You also want to put the least cost checks first especially in an operation where you're potentially looping many entities. It's much cheaper to check if an NPC is a guard or an NPC at all and not process distance checks at all than it is to run a bunch of distance checks regardless of whether you care about the NPC or not and then run your attribute checking

Example

```cpp
	for (auto &e : entity_list.GetCloseMobList(spell_target)) {
		auto mob = e.second;

		// check for NPC and also cast to NPC after successful validation that it's an NPC
		if (mob->IsNPC() && mob->CastToNPC()->IsGuard()) {

			float distance = Distance(spell_target->CastToClient()->m_Position, mob->GetPosition());
			if (mob->CheckLosFN(spell_target) && distance <= 70 || mob->CheckLosFN(this) && distance <= 70) {
				// ...
			}

		}
	}
```